lib/routes/ceph/blog.ts

```diff
@@ -34,41 +34,45 @@ export const route: Route = {
 async function handler(ctx: Context): Promise<Data> {
     const { category } = ctx.req.param();
     const url = category ? `https://ceph.io/en/news/blog/category/${category}/` : 'https://ceph.io/en/news/blog/';
-    const response = await got.get(url);
-    const data = response.data;
-    const $ = load(data);
-    const list = $('#main .section li')
-        .toArray()
-        .map((e) => {
-            const element = $(e);
-            const title = element.find('a').text().trim();
-            const link = element.find('a').attr('href');
-            const pubDate = parseDate(element.find('time').attr('datetime'));
+    return await cache.tryGet(
+        url,
+        async () => {
+            const response = await got.get(url);
+            const data = response.data;
+            const $ = load(data);
+            const list = $('#main .section li')
+                .toArray()
+                .map((e) => {
+                    const element = $(e);
+                    const title = element.find('a').text().trim();
+                    const pubDate = parseDate(element.find('time').attr('datetime'));
+                    return {
+                        title,
+                        link: new URL(element.find('a').attr('href'), 'https://ceph.io').href,
+                        pubDate,
+                    };
+                });
 
-            return {
-                title,
-                link: 'https://ceph.io/' + link,
-                pubDate,
-            };
-        });
+            const result = await Promise.all(
+                list.map((item) =>
+                    cache.tryGet(item.link, async () => {
+                        const itemReponse = await got.get(item.link);
+                        const data = itemReponse.data;
+                        const item$ = load(data);
 
-    const result = await Promise.all(
-        list.map((item) =>
-            cache.tryGet(item.link, async () => {
-                const itemReponse = await got.get(item.link);
-                const data = itemReponse.data;
-                const item$ = load(data);
+                        item.author = item$('#main section > div:nth-child(1) span').text().trim();
+                        item.description = item$('#main section > div:nth-child(2) > div').html();
+                        return item;
+                    })
+                )
+            );
 
-                item.author = item$('#main section > div:nth-child(1) span').text().trim();
-                item.description = item$('#main section > div:nth-child(2) > div').html();
-                return item;
-            })
-        )
+            return {
+                title: 'Ceph Blog',
+                link: url,
+                item: result,
+            };
+        },
+        3600
```

Comment:
No magic number here. RSSHub will cache the whole response for `CACHE_EXPIRE` seconds. If you really want to cache the index, please break the cache apart instead of nesting.

Reply:
thx, remove the outside cache